src/dos/cdrom.h

```diff
@@ -472,7 +472,41 @@ class CDROM_Interface_Ioctl : public CDROM_Interface_Physical {
 	int cdrom_fd = -1;
 };
 
-#endif /* LINUX */
+#elif defined(WIN32)
+
+#include <windows.h>
+
+class CDROM_Interface_Win32 : public CDROM_Interface_Physical {
+public:
+	~CDROM_Interface_Win32() override;
+
+	bool SetDevice(const char* path) override;
+	bool GetUPC(unsigned char& attr, std::string& upc) override;
+	bool GetAudioTracks(uint8_t& stTrack, uint8_t& end, TMSF& leadOut) override;
+	bool GetAudioTrackInfo(uint8_t track, TMSF& start, unsigned char& attr) override;
+	bool GetAudioSub(unsigned char& attr, unsigned char& track,
+	                 unsigned char& index, TMSF& relPos, TMSF& absPos) override;
+	bool GetMediaTrayStatus(bool& mediaPresent, bool& mediaChanged,
+	                        bool& trayOpen) override;
+	bool ReadSectors(PhysPt buffer, const bool raw, const uint32_t sector,
+	                 const uint16_t num) override;
+	bool ReadSectorsHost(void* buffer, bool raw, unsigned long sector,
+	                     unsigned long num) override;
+	bool LoadUnloadMedia(bool unload) override;
```

Comment:
Probably best to remove `const` for non-reference and non-pointer arguments from the function _declarations_ (but keep them in the function _definitions_).

Reply:
Sorry @weirddan455, can you drop that commit as per our recent discussion?

Reply:
Commit dropped